Pauline/feat/3 condition1

paulinegarelli opened the pull request.

Commit:
feat: add angle method for Points #38

New method to compute the angle formed by three points.

Commit:
fix: merged changes from main

Commit:
feat: Radius of the smallest circle containing three points #39

Commit:
feat: Condition 1 method #3

hampfh commented on decide/src/main/java/org/example/Decide.java:
```diff
@@ -26,7 +26,10 @@ public boolean condition0() {
 
     public boolean condition1() {
-        return true;
+        return IntStream.range(0, settings.NUMPOINTS - 2).anyMatch(
```
👍

hampfh commented on decide/src/main/java/org/example/Point.java:
```diff
@@ -16,4 +16,45 @@ public double distance(Point other) {
+     * the smallest circle containing the triangle has the opposite side as diameter
+     * Else:
+     * the smallest circle is the circumscribed circle.
+     * the radius is computed thanks to the law of sines : R = a * b * c / (4 * S)
```
From this comment it looks like the formula belongs to the law of sines, but from what I can tell this is the formula for a circumscribed circle. Might be worth just clarifying this

paulinegarelli replied:
Yes, what I meant is that we use the law of sines to compute the radius of the circumscribed circle but you're right I'll change that

hampfh commented on decide/src/main/java/org/example/Point.java:
```diff
@@ -16,4 +16,45 @@ public double distance(Point other) {
+     */
+    public static double smallestCircleRadius(Point p1, Point p2, Point p3) {
+        // checking if any of the angles is obtuse
+        if (p1.angle(p2, p3) > Math.PI) {
```
If you want to check if the angle is obtuse shouldn't you check `> Math.PI / 2`?

paulinegarelli replied:
Right, thanks!

hampfh reviewed (changes requested):
Nice work overall, I have one question regarding a comment and one thing that I think is a bug (please correct me if I'm wrong)

Commit:
fix: Wrong angle used to compare in smallestCircleRadius #42

Change the angle to PI/2
Change the comment on the computation of the radius of the circumscribed circle

paulinegarelli committed:
Merge branch 'main' into pauline/feat/3-condition1

hampfh reviewed (approved):
Thank you for making the changes, great work!
